### PR TITLE
Don't set number of replicas for indices that rely on auto_expand_replicas

### PR DESCRIPTION
**Pull request by javanna**
This is a follow-up of #108994. 

Many of our internal indices that rely on auto_expand_replicas get created providing a manual number of replicas. Such number will be immediately overridden by the auto expand replicas functionality according to the number of data nodes available. While this causes no harm, it seems misleading and unnecessary, a potential misuse that we can avoid for indices that we create ourselves.

Ideally we'd even prevent this from happening by rejecting such index creation requests, but that would be a breaking change that we'd prefer not to make at this time.

This PR removes setting number of replicas from json files (mostly templates) when auto_expand_replicas is set at the same time while creating an index

### CODE REVIEW

**Review by miltonhultgren — CHANGES_REQUESTED** [review]
For the Stack Monitoring mappings, we also need to increment the registry version, as seen in this PR https://github.com/elastic/elasticsearch/pull/94062/files

**Comment by javanna** [comment]
Thanks @miltonhultgren , done.

**Review by miltonhultgren — APPROVED** [review]

File: x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/MonitoringTemplateRegistry.java [hunk 1]
```diff
@@ -77,7 +77,7 @@ public class MonitoringTemplateRegistry extends IndexTemplateRegistry {
      * writes monitoring data in ECS format as of 8.0. These templates define the ECS schema as well as alias fields for the old monitoring
      * mappings that point to the corresponding ECS fields.
      */
-    public static final int STACK_MONITORING_REGISTRY_VERSION = 8_00_00_99 + 16;
+    public static final int STACK_MONITORING_REGISTRY_VERSION = 8_00_00_99 + 17;
```
Comment (prdoyle) [inline x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/MonitoringTemplateRegistry.java:80]: Is this the only such change required?

**Review by prdoyle — APPROVED** [review]
LGTM. One minor question.

**Review by rjernst — APPROVED** [review]
LGTM too

File: x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/MonitoringTemplateRegistry.java (hunk 1 above)
Comment (javanna) [reply x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/MonitoringTemplateRegistry.java:80]: As far as I know:) that's what Milton brought up above.

**Comment by javanna** [comment]
Thanks all for the reviews!

**Pull request merged**